gcp_airflow_foundations/source_class/source.py

```diff
@@ -54,60 +45,129 @@ def build_dags(self):
         data_source = self.config.source
         logging.info(f"Building DAG for {data_source.name}")
 
+        table_names_to_ingest = [self.config.tables[i].table_name for i in range(len(self.config.tables))]
+        templated_config = False
+        if data_source.regex_matching:
+            full_ingestion_options = data_source.full_ingestion_options
+            if full_ingestion_options.dag_creation_mode == "SOURCE":
+                return self.build_dags_source_level()
+            else:
+                table_names_to_ingest = self.get_source_tables_to_ingest()
+                templated_table_config = self.config.tables[0]
+                templated_config = True
```

Comment:
I would move it to line 51. Just for readability. templated_config = True because 'data_source.regex_matching' is True, not because full_ingestion_options.dag_creation_mode != "SOURCE"